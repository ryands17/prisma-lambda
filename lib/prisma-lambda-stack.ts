import { join } from 'path'
import * as cdk from '@aws-cdk/core'
import * as ln from '@aws-cdk/aws-lambda-nodejs'
import * as lambda from '@aws-cdk/aws-lambda'

const lambdaFn = join(__dirname, '..', 'functions', 'index.ts')

export class PrismaLambdaStack extends cdk.Stack {
  constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props)

    // The code that defines your stack goes here
    new ln.NodejsFunction(this, 'prisma', {
      runtime: lambda.Runtime.NODEJS_14_X,
      handler: 'handler',
      entry: lambdaFn,
      timeout: cdk.Duration.seconds(10),
      memorySize: 1024,
      environment: {
        DB_URL: process.env.DB_URL || '',
      },
      bundling: {
        nodeModules: ['@prisma/client', 'prisma'],
        commandHooks: {
          beforeBundling(inputDir: string, outputDir: string): string[] {
            return []
          },
          beforeInstall(inputDir: string, outputDir: string) {
            return [`cp -R ../prisma ${outputDir}/`]
          },
          afterBundling(inputDir: string, outputDir: string): string[] {
            return [
              `cd ${outputDir}`,
              `yarn prisma generate`,
              `rm -rf node_modules/@prisma/engines`,
              `rm -rf node_modules/@prisma/client/node_modules node_modules/.bin node_modules/prisma`,
            ]
          },
        },
      },
    })
  }
}
