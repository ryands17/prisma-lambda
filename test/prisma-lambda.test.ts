import { SynthUtils } from '@aws-cdk/assert'
import * as cdk from '@aws-cdk/core'
import * as PrismaLambda from '../lib/prisma-lambda-stack'

test('Snapshot matches correctly', () => {
  const stack = createStack({ outdir: 'cdk.out', stackTraces: false })
  const snapshot = SynthUtils.synthesize(stack)
  expect(snapshot).toMatchSnapshot()
})

function createStack(props?: cdk.AppProps) {
  const app = new cdk.App(props)
  return new PrismaLambda.PrismaLambdaStack(app, 'MyTestStack')
}
