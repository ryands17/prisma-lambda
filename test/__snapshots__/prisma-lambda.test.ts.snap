// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot matches correctly 1`] = `
CloudFormationStackArtifact {
  "_dependencyIDs": Array [],
  "_deps": Array [],
  "assembly": CloudAssembly {
    "artifacts": Array [
      [Circular],
      TreeCloudArtifact {
        "_dependencyIDs": Array [],
        "_deps": Array [],
        "assembly": [Circular],
        "file": "tree.json",
        "id": "Tree",
        "manifest": Object {
          "properties": Object {
            "file": "tree.json",
          },
          "type": "cdk:tree",
        },
        "messages": Array [],
      },
    ],
    "directory": "cdk.out",
    "manifest": Object {
      "artifacts": Object {
        "MyTestStack": Object {
          "environment": "aws://unknown-account/unknown-region",
          "metadata": Object {
            "/MyTestStack": Array [
              Object {
                "data": Object {
                  "artifactHashParameter": "AssetParametersc266b5913a00b6d343353a7670199b43f43c651f7fa0736d6eb34ba9eb1ca331ArtifactHashAA72AEA4",
                  "id": "c266b5913a00b6d343353a7670199b43f43c651f7fa0736d6eb34ba9eb1ca331",
                  "packaging": "zip",
                  "path": "asset.c266b5913a00b6d343353a7670199b43f43c651f7fa0736d6eb34ba9eb1ca331",
                  "s3BucketParameter": "AssetParametersc266b5913a00b6d343353a7670199b43f43c651f7fa0736d6eb34ba9eb1ca331S3Bucket3B7F916E",
                  "s3KeyParameter": "AssetParametersc266b5913a00b6d343353a7670199b43f43c651f7fa0736d6eb34ba9eb1ca331S3VersionKey27045FC8",
                  "sourceHash": "c266b5913a00b6d343353a7670199b43f43c651f7fa0736d6eb34ba9eb1ca331",
                },
                "type": "aws:cdk:asset",
              },
            ],
            "/MyTestStack/AssetParameters/c266b5913a00b6d343353a7670199b43f43c651f7fa0736d6eb34ba9eb1ca331/ArtifactHash": Array [
              Object {
                "data": "AssetParametersc266b5913a00b6d343353a7670199b43f43c651f7fa0736d6eb34ba9eb1ca331ArtifactHashAA72AEA4",
                "type": "aws:cdk:logicalId",
              },
            ],
            "/MyTestStack/AssetParameters/c266b5913a00b6d343353a7670199b43f43c651f7fa0736d6eb34ba9eb1ca331/S3Bucket": Array [
              Object {
                "data": "AssetParametersc266b5913a00b6d343353a7670199b43f43c651f7fa0736d6eb34ba9eb1ca331S3Bucket3B7F916E",
                "type": "aws:cdk:logicalId",
              },
            ],
            "/MyTestStack/AssetParameters/c266b5913a00b6d343353a7670199b43f43c651f7fa0736d6eb34ba9eb1ca331/S3VersionKey": Array [
              Object {
                "data": "AssetParametersc266b5913a00b6d343353a7670199b43f43c651f7fa0736d6eb34ba9eb1ca331S3VersionKey27045FC8",
                "type": "aws:cdk:logicalId",
              },
            ],
            "/MyTestStack/prisma/Resource": Array [
              Object {
                "data": "prismaD328B0A1",
                "type": "aws:cdk:logicalId",
              },
            ],
            "/MyTestStack/prisma/ServiceRole/Resource": Array [
              Object {
                "data": "prismaServiceRoleC66F7E04",
                "type": "aws:cdk:logicalId",
              },
            ],
          },
          "properties": Object {
            "templateFile": "MyTestStack.template.json",
          },
          "type": "aws:cloudformation:stack",
        },
        "Tree": Object {
          "properties": Object {
            "file": "tree.json",
          },
          "type": "cdk:tree",
        },
      },
      "version": "9.0.0",
    },
    "runtime": Object {
      "libraries": Object {},
    },
    "version": "9.0.0",
  },
  "assets": Array [
    Object {
      "artifactHashParameter": "AssetParametersc266b5913a00b6d343353a7670199b43f43c651f7fa0736d6eb34ba9eb1ca331ArtifactHashAA72AEA4",
      "id": "c266b5913a00b6d343353a7670199b43f43c651f7fa0736d6eb34ba9eb1ca331",
      "packaging": "zip",
      "path": "asset.c266b5913a00b6d343353a7670199b43f43c651f7fa0736d6eb34ba9eb1ca331",
      "s3BucketParameter": "AssetParametersc266b5913a00b6d343353a7670199b43f43c651f7fa0736d6eb34ba9eb1ca331S3Bucket3B7F916E",
      "s3KeyParameter": "AssetParametersc266b5913a00b6d343353a7670199b43f43c651f7fa0736d6eb34ba9eb1ca331S3VersionKey27045FC8",
      "sourceHash": "c266b5913a00b6d343353a7670199b43f43c651f7fa0736d6eb34ba9eb1ca331",
    },
  ],
  "assumeRoleArn": undefined,
  "bootstrapStackVersionSsmParameter": undefined,
  "cloudFormationExecutionRoleArn": undefined,
  "displayName": "MyTestStack",
  "environment": Object {
    "account": "unknown-account",
    "name": "aws://unknown-account/unknown-region",
    "region": "unknown-region",
  },
  "id": "MyTestStack",
  "manifest": Object {
    "environment": "aws://unknown-account/unknown-region",
    "metadata": Object {
      "/MyTestStack": Array [
        Object {
          "data": Object {
            "artifactHashParameter": "AssetParametersc266b5913a00b6d343353a7670199b43f43c651f7fa0736d6eb34ba9eb1ca331ArtifactHashAA72AEA4",
            "id": "c266b5913a00b6d343353a7670199b43f43c651f7fa0736d6eb34ba9eb1ca331",
            "packaging": "zip",
            "path": "asset.c266b5913a00b6d343353a7670199b43f43c651f7fa0736d6eb34ba9eb1ca331",
            "s3BucketParameter": "AssetParametersc266b5913a00b6d343353a7670199b43f43c651f7fa0736d6eb34ba9eb1ca331S3Bucket3B7F916E",
            "s3KeyParameter": "AssetParametersc266b5913a00b6d343353a7670199b43f43c651f7fa0736d6eb34ba9eb1ca331S3VersionKey27045FC8",
            "sourceHash": "c266b5913a00b6d343353a7670199b43f43c651f7fa0736d6eb34ba9eb1ca331",
          },
          "type": "aws:cdk:asset",
        },
      ],
      "/MyTestStack/AssetParameters/c266b5913a00b6d343353a7670199b43f43c651f7fa0736d6eb34ba9eb1ca331/ArtifactHash": Array [
        Object {
          "data": "AssetParametersc266b5913a00b6d343353a7670199b43f43c651f7fa0736d6eb34ba9eb1ca331ArtifactHashAA72AEA4",
          "type": "aws:cdk:logicalId",
        },
      ],
      "/MyTestStack/AssetParameters/c266b5913a00b6d343353a7670199b43f43c651f7fa0736d6eb34ba9eb1ca331/S3Bucket": Array [
        Object {
          "data": "AssetParametersc266b5913a00b6d343353a7670199b43f43c651f7fa0736d6eb34ba9eb1ca331S3Bucket3B7F916E",
          "type": "aws:cdk:logicalId",
        },
      ],
      "/MyTestStack/AssetParameters/c266b5913a00b6d343353a7670199b43f43c651f7fa0736d6eb34ba9eb1ca331/S3VersionKey": Array [
        Object {
          "data": "AssetParametersc266b5913a00b6d343353a7670199b43f43c651f7fa0736d6eb34ba9eb1ca331S3VersionKey27045FC8",
          "type": "aws:cdk:logicalId",
        },
      ],
      "/MyTestStack/prisma/Resource": Array [
        Object {
          "data": "prismaD328B0A1",
          "type": "aws:cdk:logicalId",
        },
      ],
      "/MyTestStack/prisma/ServiceRole/Resource": Array [
        Object {
          "data": "prismaServiceRoleC66F7E04",
          "type": "aws:cdk:logicalId",
        },
      ],
    },
    "properties": Object {
      "templateFile": "MyTestStack.template.json",
    },
    "type": "aws:cloudformation:stack",
  },
  "messages": Array [],
  "name": "MyTestStack",
  "originalName": "MyTestStack",
  "parameters": Object {},
  "requiresBootstrapStackVersion": undefined,
  "stackName": "MyTestStack",
  "stackTemplateAssetObjectUrl": undefined,
  "tags": Object {},
  "templateFile": "MyTestStack.template.json",
  "terminationProtection": undefined,
}
`;
